pkgs/build-support/mkfont.nix

```diff
@@ -0,0 +1,60 @@
+{ lib, stdenvNoCC }:
+
+args:
+
+let
+  noUnpackFonts = lib.hasAttr "noUnpackFonts" args && args.noUnpackFonts;
+  hasMultipleSources = lib.hasAttr "srcs" args;
+in
+stdenvNoCC.mkDerivation ({
+  dontPatch = true;
+  dontConfigure = true;
+  dontBuild = true;
+  doCheck = false;
+  dontFixup = true;
+
+  sourceRoot = if noUnpackFonts || hasMultipleSources then "." else null;
+  unpackCmd = if noUnpackFonts then ''
+    filename="$(basename "$(stripHash "$curSrc")")"
+    cp $curSrc "./$filename"
+  '' else null;
+
+  /*
+  installPhase = ''
+    # these come up in some source trees, but are never useful to us
+    find -iname __MACOSX -type d -print0 | xargs -0 rm -rf
+    # find -type f,l
+
+    for pattern in "*.ttf" "*.ttc" "*.ttf" "*.otf" "*.eot" "*.bdf" "*.otb" "*.psf"; do
+      find -ipath "$pattern" | while IFS= read -r filename; do
+        case "$(basename "$filename")" in
+          *.ttf) fontdir=$out/share/fonts/truetype/ ;;
+          *.ttc) fontdir=$out/share/fonts/truetype/ ;;
+          *.ttf) fontdir=$out/share/fonts/truetype/ ;;
+          *.otf) fontdir=$out/share/fonts/opentype/ ;;
+          *.eot) fontdir=$out/share/fonts/eot/ ;;
+          *.bdf) fontdir=$out/share/fonts/misc/ ;;
+          *.otb) fontdir=$out/share/fonts/misc/ ;;
+          *.psf) fontdir=$out/share/consolefonts/ ;;
+          *) echo "File has invalid extension: $filename"; exit 1 ;;
+        esac
+
+        install -v -m644 --target $fontdir -D "$filename"
+      done
+    done
+  '';
+  */
+
+  installPhase = ''
```

Comment:
This version of `installPhase` doesn't install `.eot` fonts (Your diff/gist shows at least one package where the `.eot` files are no longer there. The commented out `installPhase` above does include `.eot`, so I assume this is an accidental omission).

Reply:
It's not, actually :) `eot` files are supported in Internet Explorer 6 to 11, but [no other browser whatsoever](https://caniuse.com/#feat=eot). IE9 (released 9 years ago) and upwards support OTF, TTF, and WOFF, so the only reason to ever ship eot files is compatibility with IE6 to IE8. I suspect that the packages that include `eot` files don't do so out of careful consideration.

I propose we omit `eot` files from nix packaging.

(Of course, decisions such as this should be annotated in `mkFont` to make it easy to discover for packagers who look up how things work, and also in the manual. The "needs documentation" tag definitely applies to PR, I'll try to get around to it so we have a better basis for discussion soon ;)

Reply:
Looks like `.woff` and `.woff2` fonts aren't handled either.